[wilsonwid] iP

wilsonwid opened the pull request:
# TalkingBot
> TalkingBot is a CLI-based application designed to aid individuals in remembering their daily tasks.

 TalkingBot allows allows for three task types:
- Todos
- Deadlines
- Events

Some other features include:
1. Date formatting 📆
2. Task saving to disk 💾
3. Marking/unmarking your tasks ✅
4. Deleting tasks 🗑️

**Task list:**
- [ ] Add more customisation to the bot
- [ ] Create more features

Included below is the `TalkingBot` class  in Java:

```java
public class TalkingBot {
    // content of the class
}
```

Note that this is a coursework project for CS2103T Software Engineering. The course website can be found [here](https://nus-cs2103-ay2324s2.github.io/website/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Add implementation for Level-0

Commit:
Added Add.java, Command.java

Commit:
Added echo

Commit:
Add horizonal line, Task.java, and TaskList.java

Commit:
Add horizontal line, tabs, and proper newlines

Commit:
Add marking

Commit:
Add minor changes to formatting

Commit:
Minor formatting and logic fixes

Commit:
Add Todo.java, create additional character for bot

Commit:
Add Deadline.java and Duke.java, added formatting fixes

Commit:
Added input, fixed cases

Commit:
Add test case and modified Duke.java

Commit:
Add more testcases, add DukeException.java

Commit:
Add several errors

Commit:
Add regex matching for todo

Commit:
Add deletion

Commit:
Add deletions

Commit:
Modified to use the Type enum

Commit:
Removed redundant iterator check variable

Commit:
Add SaveFile.java, modify Task.java and Duke.java

Commit:
Change Type to TaskType, add new functionality to get tasks from and save tasks to file

Commit:
Delete redundant constructor

Commit:
Merge branch 'branch-Level-7'

Commit:
Add more exceptions and additional functionality

Commit:
Modify Duke to TalkingBot, fix bugs

Commit:
Merge branch 'branch-Level-7'

Commit:
Add LocalDateTime

Commit:
Fix issue with date parsing

Commit:
Add formatting fixes to time

Commit:
Fix saving format issue

Commit:
Fix marking issue

Commit:
Merge branch 'branch-Level-8'

Commit:
Add skeleton code for level

Commit:
Made Command classes, update code

Commit:
Add more ui elements

Commit:
Move printing functionality to Ui

Commit:
Add goodbye message

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Move files into packages

Commit:
Move TalkingBot.java

Commit:
Sort imports lexicographically

Commit:
Sort imports lexicographically

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into add-gradle-support

Commit:
Merge branch 'add-gradle-support'

Commit:
Made changes to build.gradle

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Add JUnit tests

Commit:
Merge branch 'A-JUnit'

Commit:
Create some Javadoc comments

Commit:
Add Javadoc comments for more classes

Commit:
Add more Javadoc comments

Commit:
Add more Javadocs

Commit:
Add more Javadocs

Commit:
Add Javadoc to Ui.java

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Reorder import statements

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add FindCommand.java and filter functionality

Commit:
Add to Parser.java

Commit:
Fix merge conflict

Commit:
Add Javadoc comments

Commit:
Fix bug in tasks

Commit:
Add to gradle build

Commit:
Add checkstyle and fix violations

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Add naming changes

Commit:
Add dependencies

yleeyilin commented on src/main/java/talkingbot/task/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+     *
+     * @return Deadline ending time.
+     */
+    public LocalDateTime getEndTime() {
```
Should this getter be declared public if it is only used within the class itself?

yleeyilin reviewed (approved):
Overall, I found your code easy to read as there were good variable names used. Additionally, coding standard is well followed with clear documentation.  Just one nit with regards to declaring class methods as public.

yleeyilin reviewed (commented):
Overall, I found your code easy to read as there were good variable names used. Additionally, coding standard is well followed with clear documentation. Just one nit with regards to declaring class methods as public.

HM33-Stu commented on src/main/java/talkingbot/TalkingBot.java:
```diff
@@ -0,0 +1,55 @@
+ * Class serving as the entry point of the application.
+ */
+public class TalkingBot {
+    private static final String FILE_PATH = "./data/taskList.txt";
```
Nice! Conventions for a constant name was followed

HM33-Stu commented on src/main/java/talkingbot/command/ByeCommand.java:
```diff
@@ -0,0 +1,41 @@
+        super(commandArr);
+    }
+
+    /**
+     * Runs this command.
+     * Prints a goodbye message, saves current list to a file,
+     * and stops the running of the program.
+     *
+     * @param tasks List of tasks.
+     * @param saveFile Abstraction for a file.
+     * @param ui User interface.
+     */
```
Nice! You've used JavaDoc appropriately throughout your code

HM33-Stu commented on src/main/java/talkingbot/task/Task.java:
```diff
@@ -0,0 +1,151 @@
+            throws TalkingBotException {
+        if (type.equals("todo")) {
+            if (fullDescription.isEmpty()) {
+                throw new TalkingBotException("ERROR! todo descriptions cannot be empty"
+                        + " nor only containing whitespaces.");
```
Error messages are helpful (points out the error made by the user). Same applies for your other error messages as well

HM33-Stu commented on src/main/java/talkingbot/util/SaveFile.java:
```diff
@@ -0,0 +1,90 @@
+     * @param tasks Task list to be saved.
+     * @throws TalkingBotException If IOException occurs.
+     */
+    public void saveTasksToFile(TaskList tasks) throws TalkingBotException {
+        try {
+            FileWriter fileWriter = new FileWriter(this.saveFile);
+            StringBuilder strBuild = new StringBuilder();
+            for (int i = 0; i < tasks.getSize(); i++) {
+                Task curTask = tasks.getTask(i);
+                strBuild.append(curTask.getSaveFileString());
+                if (i < tasks.getSize() - 1) {
+                    strBuild.append("\n");
+                }
+            }
+            fileWriter.write(strBuild.toString());
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new TalkingBotException("ERROR! IOException occurred!");
```
camelCase convention is followed throughout your code

HM33-Stu commented on src/main/java/talkingbot/util/Ui.java:
```diff
@@ -0,0 +1,160 @@
+ * Class that represents the user interface.
+ */
+public class Ui {
+    private static final String H_LINE = "\t__________________________"
+            + "__________________________________";
+    private static final String WELCOME = "\tHello! I'm "
+            + "TalkingBot\n\tWhat can I do for you?";
+    private static final String GOODBYE = "\tBye. Hope to see you again soon!";
+    private static final String LOADING_ERR_MSG = "\tUsing new file instead...";
+    private static final String TASK_SET_DONE_MSG = "\tNice! I've marked this task as done:";
+    private static final String TASK_SET_UNDONE_MSG = "\tAlright, I've marked this task as undone.";
+    private static final String INVALID_CMD_MSG = "\tERROR! Unknown command detected.";
+    private static final String SAVE_ONGOING_MSG = "\tSaving tasks to file: ";
+    private static final String SAVE_DONE_MSG = "\tSave done!";
+    private static final String ADD_TASK_MSG = "\tAlright, I've added this task to your list:";
+    private static final String TASK_NUM_MSG = "\tYou now have %d tasks in the list.";
+    private static final String REMOVE_TASK_MSG = "\tRemoving task:";
```
Naming conventions for constant names are followed throughout your code.

HM33-Stu reviewed (commented):
Overall, code looks great and coding standards were followed well

Commit:
Add classes for GUI

Commit:
Add GUI components

Commit:
Modify command and other files for GUI

Commit:
Add basic GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
Add better GUI

Commit:
Update build.gradle to use new Launcher

The build.gradle file is currently using the old entry point for the CLI
application.

This commit updates the entry point to use the new Launcher.java file
that supports the GUI version of the application.

Commit:
Fix bug in file creation

The pervious version does not have a functionality for creating the
directory where the save file is to be stored in. This would cause
errors whenever such a condition occurs.

This commit fixes the error and creates a directory if there is no
directory to currently store the save file.

Commit:
Add code quality fixes

The current code uses plenty of magic literals as well as lacks the use
of constants.

This commit partially fixes the above issues by extracting out suh code
into named constants.

Commit:
Add more code quality changes

Current version of code has many magic literals, most of which have to
do with string formatting or standard displaying of messages.

This commit fixes such issues in remaining Java files.

Commit:
Add another constant

Commit:
Add assertions

Commit:
Merge pull request #1 from wilsonwid/A-Streams

Fix bug in file creation and add streams

Commit:
Merge pull request #2 from wilsonwid/branch-A-Assertions

Add assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Merge pull request #3 from wilsonwid/branch-A-CodeQuality

Add code quality fixes

Commit:
Add DoWithinPeriod and minor changes

Commit:
Add to Task and Parser

Commit:
Merge branch 'branch-B-DoWithinPeriodTasks'

Commit:
Add workflow .yml file

Commit:
Move gradle.yml file

Commit:
Fix bugs

Commit:
Add fix

Commit:
Add running changes

Commit:
Add new logic and checkstyle

Commit:
Update javadoc comments to match new checkstyle

Commit:
Delete redundant boolean

Commit:
Fix bye command not exiting

Commit:
Add welcome message and javadoc changes

Commit:
Add citation

Commit:
Modify gradle.yml and remove redirection

Commit:
Add more abstraction

Commit:
Add more abstraction and error handling

Commit:
Add more abstraction and comments

Commit:
Add better GUI

Commit:
Add javadoc for test and TodoTest

Commit:
Add DoWithinPeriodTest

Commit:
Add ParserTest

Commit:
Fix checkstyle in tests

Commit:
Add user guide

Commit:
Add user guide

Commit:
Add release tag

Commit:
Update README.md formatting

Commit:
Add more exception handling

Commit:
Delete redundant import

Commit:
Update README
